File: Recursion/a_multiple_of_three.py
import sys

cnt = 0
def recursion(val):
    global cnt
    if (len(val) < 2):
        if (not (int(val) % 3)):
            return (1)
        else:
            return (0)
    cnt += 1
    t = 0
    for i in val:
        t += int(i)
    return (recursion(str(t)))

X = sys.stdin.readline().strip()
ans = recursion(X)
print(cnt)
if (ans):
    print("YES")
else:
    print("NO")

File: Recursion/a_multiple_of_three.cpp
#include <iostream>
#include <string>
#define endl "\n"

std::string X;
int cnt;

void input_faster()
{
	std::ios_base::sync_with_stdio(false);
	std::cin.tie(0);
	std::cout.tie(0);
}

void input()
{
	input_faster();
	std::cin >> X;
}

bool recursion(std::string x)
{
	if (x.size() < 2)
	{
		int y = x[0] - '0';
		if (!(y % 3))
			return (1);
		else
			return (0);
	}
	cnt++;
	int temp = 0;
	for (size_t i = 0 ; i < x.size(); i++)
		temp += x[i] - '0';
	return(recursion(std::to_string(temp)));
}

void solve()
{
	bool x = recursion(X);
	std::cout << cnt << endl;
	if(x)
		std::cout << "YES";
	else
		std::cout << "NO";
}

int main()
{
	input();
	solve();
	return (0);
}

